internal/proxy/oauthproxy.go

```diff
@@ -602,37 +599,53 @@ func (p *OAuthProxy) isXHR(req *http.Request) bool {
 
 // SignOut redirects the request to the provider's sign out url.
 func (p *OAuthProxy) SignOut(rw http.ResponseWriter, req *http.Request) {
+	tags := []string{"action:sign_out"}
+
 	p.sessionStore.ClearSession(rw, req)
 
+	route, ok := p.router(req)
+	if !ok {
+		// this shouldn't happen since we've already matched the host once on this request
+		tags = append(tags, "error:unknown_host")
+		p.StatsdClient.Incr("application_error", tags, 1.0)
```

Comment:
I might not quite be thinking about the recorded metrics/stats correctly, but I see in the tags here we include the 'action' as well as the  'error' as opposed to in the `p.UnknownHost` function called below we only include the 'error'. However, should we only increment the `application_error` metric once per 'UnknownHost' so as to not mislead the metric data?